### ISSUE TITLE
[import/no-duplicates] false-positive "imported multiple times" in TS

### ISSUE THREAD

**Issue by echocrow** [import-js/eslint-plugin-import#2855]
We're seeing unexpected `import/no-duplicates` issues for some packages in TypeScript, which seem to happen when the different modules reuse the same, shared definitions under the hood.

For example:
```ts
import type {ActionReturn} from 'svelte/action'
import type {Readable} from 'svelte/store'
// ...
```
Expected behavior:
```
# No lint validations; these are two different, non-duplicate imports.
```
Actual behavior: 
```
  1:33  error  '{...}/node_modules/svelte/types/index.d.ts' imported multiple times  import/no-duplicates
  2:29  error  '{...}/node_modules/svelte/types/index.d.ts' imported multiple times  import/no-duplicates
```
As per the error message, I guess this is related to the fact that both imports have their types defined in a shared, private file. I'm unsure if this is an issue with the svelte packages per-se, but this being flagged by `eslint-plugin-import` seems like an unexpected bug? It may be noteworthy that the core `no-duplicate-imports` ESLint rule does _not_ flag this the same way.

Versions:
- `eslint: 8.46.0`
- `eslint-plugin-import: 2.28.0`
- `svelte: 4.1.2`

A sample repo can be found [here](https://stackblitz.com/edit/eslint-plugin-import-false-negative-no-duplicates) with the bare TypeScript & ESLint config.

**Comment by ljharb** [comment]
This is indeed because the TS resolver maps them to the same file. `no-duplicate-imports` does no resolution, so it doesn't run into this problem (but it misses many other instances as a result).

I'm traveling atm but I believe there may be an existing issue about this.

**Comment by echocrow** [comment]
> there may be an existing issue about this.

Apparently I did not search hard enough yesterday; looks this #1479 already talks about the same issue, my bad!

_I mean_
```
  #2855  warning  'issues/1479' reported multiple times  issues/no-duplicates
```

Closing as duplicate. Happy to cross-post the repro if of any value!
